auto_dev/data/templates/protocols/protodantic.jinja

```diff
@@ -0,0 +1,31 @@
+"""Module containing the pydantic models generated from the .proto file."""
+
+from __future__ import annotations
+
+from enum import IntEnum
+
+from pydantic import BaseModel
+
+from {{ primitives_import_path }} import (
+    {%- for primitive in float_primitives %}
+    {{ primitive.__name__ }},
+    {%- endfor %}
+    {%- for primitive in integer_primitives %}
+    {{ primitive.__name__ }},
+    {%- endfor %}
+)
+
+# ruff: noqa: N806, C901, PLR0912, PLR0914, PLR0915, A001
+# N806     - variable should be lowercase
+# C901     - function is too complex
+# PLR0912  - too many branches
+# PLR0914  - too many local variables
+# PLR0915  - too many statements
+# A001     - shadowing builtin names like `id` and `type`
+
+MAX_PROTO_SIZE = 2 * 1024 * 1024 * 1024  # 2 GiB in bytes
+
+{{ formatter.render(file) }}
+
+for cls in BaseModel.__subclasses__():
+    cls.model_rebuild()
```

Comment:
This is important. While `from __future__ import annotations` allows the python interpreter to work with self-referential and nested type annotations, this step is necessary for hypothesis `st.from_type` to work on our custom pydantic models. It updates forward references, circumventing any need to generated strategies for nested types and passing those to a `st.builds` of the type to generate the strategy, which would require the same kind of recursive logic as already employed to generate the pydantic models themselves. As long as the type annotation on the pydantic models is correct, `st.from_type` provides a very convenient way to generate strategies without additional code.